lib/security_response_header_filter.js

```diff
@@ -109,5 +109,10 @@ module.exports = function addCSPHeaders(hyper, req, next, options) {
         // Finally, assign the csp header variants
         cspHeaderVariants.forEach((name) => { rh[name] = csp; });
         return res;
+    };
+
+    return resPromise.then(addHeaders, (e) => {
+        e = addHeaders(e);
```

Comment:
I think we want to make sure `e` is a response here, and not an Error.

Reply:
`e` should be `HTTPError` here, shouldn't it?

Reply:
Yes, but we allow things like `throw new Error(blah)` too in the code, so it might not be what we expect it to be afaik.